radix-engine/src/blueprints/consensus_manager/validator.rs

```diff
@@ -1572,9 +1590,17 @@ impl ValidatorBlueprint {
         } else {
             amount_of_stake_units
```

Comment:
We're multiplying first and then dividing which makes this more easily prone to overflows.

I would recommend we refactor this to:

```rust
amount_of_stake_units.
    .safe_div(total_stake_unit_supply)
    .and_then(|d| d.safe_mul(active_stake_amount))
    .ok_or(ValidatorError::UnexpectedDecimalComputationError)?
```

Reply:
We also need to change `calculate_stake_unit_amount`.

Reply:
Yup, I added this to our discussion list. Whatever approach we take in pools should be the same here.